src/timer/TimerProvider.jsx

```diff
@@ -28,20 +29,16 @@ const TimerProvider = ({
   children,
 }) => {
   const { activeAttempt: attempt, exam } = useSelector(state => state.specialExams);
-  const { time_limit_mins: timeLimitMins } = exam;
   const [timeState, setTimeState] = useState({});
   const [limitReached, setLimitReached] = useToggle(false);
+  const dispatch = useDispatch();
+  const { time_limit_mins: timeLimitMins } = exam;
   const {
     desktop_application_js_url: workerUrl,
     ping_interval: pingInterval,
-    time_remaining_seconds: timeRemaining,
+    time_remaining_seconds: initSecondsLeft,
+    timer_ends: timerEnds,
   } = attempt;
-  const LIMIT = GRACE_PERIOD_SECS ? 0 - GRACE_PERIOD_SECS : 0;
-  const CRITICAL_LOW_TIME = timeLimitMins * 60 * TIME_LIMIT_CRITICAL_PCT;
-  const LOW_TIME = timeLimitMins * 60 * TIME_LIMIT_LOW_PCT;
-  let liveInterval = null;
```

Comment:
The control variable was moved to the useEffect block, since the component refresh would overwrite this value rendering it useless.